### DISCUSSION TITLE
editor-js and mdx.js

### DISCUSSION THREAD

**Discussion by ghost** [mdx-js/mdx#2130] [Q&A] [answered]
Hi all! I would like to know if all you have ever thought about joining two projects into something unique. I think two tools are essential today: **[editor-js](https://editorjs.io/)** and **mdx.js**

#### What is editor-js?
**Copyright:** "Next generation block styled editor. - editorjs.io"

**Comment by ChristianMurphy** [comment, marked as answer]
MDX is already supported in several block based editors, provided by other projects.
Such as:
* Blocks https://www.mdx-blocks.com/
* Tina https://tina.io/blog/tina-supports-mdx/
* likely more

Along with a long number of document based CMSes which support mdx.

Block based editing is outside the scope of mdx itself, and there are currently no plans to bring it in scope.

The editor.js team are welcome to support mdx as a format, as other projects have.
Though there appears to be no plans or interest from the editor js team for supporting mdx. https://github.com/codex-team/editor.js/search?q=mdx

I don't really see the projects merging or joining.
They have different goals.

**Reply by ghost** [reply]
Hi! I would like to add three reasons why this could be relevant:
1. As... well there is... the standard web components and block protocol to componentize parts of html, css and js on the web. I wish there was a universal block-to-text protocol or standardization as proposed by mdx-js, editor.js, and many others. 
2. In this sense, a partnership between project: editor.js and mdx.js would provide greater compatibility between two projects and provide a more united and integrated protocol. That is, we would have block protocol for the web and the "block-protocol: mdx.js/editor.js" for editing, standardization of text on the web.
3. The problem with the block protocol is that it doesn't support text. So providing an integration between editor.js and mdx.js could be applicable in that sense.

**Reply by ChristianMurphy** [reply]
There may be some confusion on what MDX is.

MDX is a text format, which combines markdown and JSX.
MDX includes tools which compile MDX text to JavaScript text.
The compiled JavaScript text can then be run in a number of JavaScript frameworks to produce HTML (or any other presentation format/language the framework supports).

MDX is _not_ a protocol.
MDX is _not_ an editor (though it is supported by several editors).
MDX is _not_ a web standards working group. (those already exist in [W3C](https://www.w3.org/), [WHATWG](https://whatwg.org/), and [ECMA](https://www.ecma-international.org/publications-and-standards/standards/ecma-262/). MDX aims to produce output which is compatible with these standards)

> As... well there is... the standard web components and block protocol to componentize parts of html, css and js on the web. I wish there was a universal block-to-text protocol or standardization as proposed by mdx-js, editor.js, and many others.

I'm not sure how this is relevant on a few fronts.
* Web components are already a standard, and MDX already supports web components.
* MDX is not a "block-to-text protocol", nor is it a protocol at all.
* Block editors are already welcome to adopt MDX as an intermediate text format, and several do.

> In this sense, a partnership between project: editor.js and mdx.js would provide greater compatibility between two projects and provide a more united and integrated protocol. That is, we would have block protocol for the web and the "block-protocol: mdx.js/editor.js" for editing, standardization of text on the web.

The same points as above apply here.
I'd additionally note that the MDX community is happy to answer questions from, and offer pointers to, adopters. Including editor js if they choose to support MDX as an output.
It would be more a question of if editor js is interested in supporting MDX as an output format.

> The problem with the block protocol is that it doesn't support text. So providing an integration between editor.js and mdx.js could be applicable in that sense.

I'm not sure what the "block protocol" is.
If you mean editor js' output, that is already (JSON) text, from their docs:

> It returns clean data output in JSON

(source: https://editorjs.io/)
And that JSON text could be translated to MDX text, by editor js itself, or by a community project created by you.

**Reply by ChristianMurphy** [reply]
To summarize:
* If you're interested in creating an integration between editor js and MDX, feel free to ask specific questions, the MDX community is supportive of adopters/integrators.
* If you're interested in MDX merging with editor js, that is unlikely, these are two very different projects with different goals.
* If you are interested in getting all editing platforms on to a single format. The best route would probably be to form a standards working group under an existing standards body, and persuade the editor projects to join.

**Reply by ghost** [reply]
Hi ChristianMurphy. Thank you so much for answering all my doubts.
